compiler/damlc/tests/daml-test-files/Daml3ScriptTrySubmitConcurrentlyKeys.daml

```diff
@@ -0,0 +1,31 @@
+-- Copyright (c) 2023 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+-- @ SCRIPT-V2
+
+{-# LANGUAGE ApplicativeDo #-}
+
+module Daml3ScriptTrySubmitConcurrentlyKeys where
+
+import DA.Assert ((===))
+import DA.Either (partitionEithers)
+import Daml.Script
+
+template T
+  with
+    p : Party
+  where
+    signatory p
+    key p : Party
+    maintainer key
+
+main : Script ()
+main = do
+  alice <- allocateParty "alice"
+  rs <- alice `trySubmitConcurrently` do
+    replicate n (createCmd T with p = alice)
+  let (lefts, rights) = partitionEithers rs
+  length lefts === n - 1
+  length rights === 1
```

Comment:
this fails, which means there's something very wrong in my implementation. I'll investigate tomorrow